pkg/urbit/vere/io/ames.c

```diff
@@ -446,38 +679,149 @@ _ames_recv_cb(uv_udp_t*        wax_u,
 {
   u3_ames* sam_u = wax_u->data;
 
-  //  data present, and protocol version in header matches 0
+  c3_o pas_o = c3y;
+
+  //  ensure a sane message size
   //
-  //    XX inflexible, scry version out of ames
+  if (4 >= nrd_i) {
+    pas_o = c3n;
+  }
+
+  //  ensure the protocol version matches ours
   //
-  if (  (0 < nrd_i)
-     && (0 == (0x7 & *((c3_w*)buf_u->base))) )
+  if ( c3y == pas_o
+    && (c3y == sam_u->fit_o)
+    && (sam_u->ver_y != (0x7 & *((c3_w*)buf_u->base))) )
```

Comment:
The packet header should be parsed prior to this point, and this check should be performed against the version in the resulting `u3_head` struct.